tests/acceptance/features/bootstrap/GraphContext.php

```diff
@@ -751,4 +765,92 @@ public function theLastResponseShouldBeUnauthorizedReponse(): void {
 			. "\nExpected unauthorized message but got '" . $errorText . "'"
 		);
 	}
+
+	/**
+	 * @param string $user
+	 *
+	 * @return ResponseInterface
+	 * @throws JsonException
+	 * @throws GuzzleException
+	 */
+	public function retrieveUserInformationUsingGraphApi(
+		string $user
+	):ResponseInterface {
+		$credentials = $this->getAdminOrUserCredentials($user);
+		return GraphHelper::getUserInformation(
+			$this->featureContext->getBaseUrl(),
+			$this->featureContext->getStepLineRef(),
+			$credentials["username"],
+			$credentials["password"],
+		);
+	}
+
+	/**
+	 * @When /^the user "([^"]*)" retrives (:?her|his) information using the Graph API$/
+	 *
+	 * @param string $user
+	 *
+	 * @return void
+	 * @throws JsonException
+	 */
+	public function userRetrievesHisorHerInformationOfUserUsingGraphApi(
+		string $user
+	):void {
+		$response = $this->retrieveUserInformationUsingGraphApi($user);
+		$this->featureContext->setResponse($response);
+	}
+
+	/**
+	 * @Then /^the api response should contains the following information:$/
+	 *
+	 * @param TableNode $table
+	 *
+	 * @return void
+	 */
+	public function theApiResponseForUserShouldContainsTheFollowingInformation(TableNode $table): void {
+		$rows = $table->getRowsHash();
+		$apiResponse = \json_decode((string)$this->featureContext->getResponse()->getBody(), true, 512, JSON_THROW_ON_ERROR);
+		// assertion of the user is member of groups
+		if ($rows['memberOf']) {
+			// collect memberOf from response
+			$memberOfFromApiReponse = [];
+			$memberOf = preg_split('/\s*,\s*/', trim($rows['memberOf']));
+			foreach ($apiResponse['memberOf'] as $member) {
+				$memberOfFromApiReponse[] = $member['displayName'];
+			}
+			Assert::assertEqualsCanonicalizing($memberOf, $memberOfFromApiReponse);
+		}
+		// check if the user_if from response is in format UUIDv4
+		$isUUIDv4 = $this->featureContext->substituteInLineCodes(
+			$rows['id'],
+			$this->featureContext->getCurrentUser(),
+			[],
+			[
+				[
+					"code" => "%UUIDv4%",
+					"function" =>
+						[$this, "checkUUIDv4PatternForUserId"],
+					"parameter" => [$apiResponse['id']]
+				],
+			]
+		);
+		Assert::assertEquals(
+			1,
+			$isUUIDv4,
+			__METHOD__ .
+			$apiResponse['id'] . ' ID is not in the format of UUIDv4'
+		);
```

Comment:
Could you move this inside the foreach block below? Do all assertions there
If we don't want to check to `id`, this will probably fail.